src/pkg/path/path.go

```diff
@@ -317,6 +338,60 @@ func Split(segment string) []string {
 	return res
 }
 
+func ArePathsEquivalent(path1, path2 string) bool {
+	normalizedPath1 := strings.TrimSpace(filepath.Clean(path1))
+	normalizedPath2 := strings.TrimSpace(filepath.Clean(path2))
+
+	normalizedPath1 = strings.TrimSuffix(normalizedPath1, string(filepath.Separator))
+	normalizedPath2 = strings.TrimSuffix(normalizedPath2, string(filepath.Separator))
+
+	return normalizedPath1 == normalizedPath2
+}
+
+func IsValidPath(path string, fs FileSystem) bool {
+	isDir := strings.TrimSpace(filepath.Ext(path)) == ""
```

Comment:
That feels sketchy. I feel like we should allow them to have filenames without an extension.